internal/core/application/feeder/service.go

```diff
@@ -0,0 +1,175 @@
+package feeder
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/tdex-network/tdex-daemon/internal/core/domain"
+
+	"github.com/tdex-network/tdex-daemon/internal/core/ports"
+)
+
+type Service struct {
+	repoManager ports.RepoManager
+	feederSvc   ports.PriceFeeder
+}
+
+func NewService(
+	feederSvc ports.PriceFeeder,
+	repoManager ports.RepoManager,
+) (Service, error) {
+	svc := Service{
+		repoManager: repoManager,
+		feederSvc:   feederSvc,
+	}
+
+	if err := svc.startPriceFeed(feederSvc); err != nil {
+		return Service{}, err
+	}
+
+	return svc, nil
+}
+
+func (s *Service) Stop(ctx context.Context) {
+	s.feederSvc.Stop(ctx)
+}
+
+func (s *Service) StartFeed(ctx context.Context, feedID string) error {
+	priceFeed, err := s.feederSvc.GetPriceFeed(ctx, feedID)
+	if err != nil {
+		return err
+	}
+
+	market, err := s.repoManager.MarketRepository().GetMarketByAssets(
+		ctx,
+		priceFeed.GetMarket().GetBaseAsset(),
+		priceFeed.GetMarket().GetQuoteAsset(),
+	)
+	if err != nil {
+		return err
+	}
+
+	if !market.IsStrategyPluggable() {
+		return fmt.Errorf("market is not pluggable")
+	}
+
+	return s.feederSvc.StartFeed(ctx, feedID)
+}
+
+func (s *Service) StopFeed(ctx context.Context, feedID string) error {
+	return s.feederSvc.StopFeed(ctx, feedID)
+}
+
+func (s *Service) AddPriceFeed(
+	ctx context.Context, market ports.Market, source, ticker string,
+) (string, error) {
+	return s.feederSvc.AddPriceFeed(ctx, market, source, ticker)
+}
```

Comment:
let's check if the market really exists:
```suggestion
) (string, error) {
        if _, err := s.repoManager.MarketRepository().GetMarketByAssets(
		ctx,
		priceFeed.GetMarket().GetBaseAsset(),
		priceFeed.GetMarket().GetQuoteAsset(),
	); err != nil {
	        return "", err
	}
	return s.feederSvc.AddPriceFeed(ctx, market, source, ticker)
}
```

Reply:
Done